client/src/app-components/ticket-viewer.js

```diff
@@ -83,30 +83,45 @@ class TicketViewer extends React.Component {
 
     render() {
         const { ticket, userStaff, userId, editable, allowAttachments, assignmentAllowed } = this.props;
-        const showResponseField = (!ticket.closed && (editable || !assignmentAllowed));
+        const { editTitle, loading, edit, editId } = this.state;
+        const { closed, author, content, date, edited, file, events} = ticket;
+        const showResponseField = (!closed && (editable || !assignmentAllowed));
+        const eventsWithModifiedComments = [...events];
+        let lastCommentIndex = 0;
+
+        for(let index = eventsWithModifiedComments.length-1; 0 < index; index--) {
+            if(eventsWithModifiedComments[index].type === "COMMENT" && lastCommentIndex < index) {
+                lastCommentIndex = index;
+                eventsWithModifiedComments[index].isLastComment = true;
+            } else {
+                eventsWithModifiedComments[index].isLastComment = false;
+            }
+        }
```

Comment:
We want to avoid changes to something that's already declared as much as possible, the idea for that is to try hard to use `const` everywhere, and in case of arrays/objects, to try to avoid modifying their properties/elements.

I suggest the following code for this part:

```typescript
const eventsWithModifiedComments = events.map((event, index) => {
  return {...event, isLastComment: index === events.length - 1 && event.type === "COMMENT"};
}
```

But please adapt it to the style of the codebase, and of course check if it's correct and does the same thing you want.